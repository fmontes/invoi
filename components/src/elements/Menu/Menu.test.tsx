import * as React from 'react';
import { Menu } from './index';
import { shallow } from 'enzyme';

test('Menu should render', () => {
    const component = shallow(
        <Menu
            theme={{}}
            options={[
                {
                    label: 'Option One',
                    onClick: (e: MouseEvent) => console.log('one')
                },
                {
                    label: 'Option Two',
                    onClick: (e: MouseEvent) => console.log('two'),
                    disabled: true
                },
                {
                    label: 'Option Three',
                    onClick: (e: MouseEvent) => console.log('three')
                }
            ]}
        />
    );
    expect(component).toMatchSnapshot();
});
