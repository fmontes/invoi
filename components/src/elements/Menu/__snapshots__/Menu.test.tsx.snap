// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render 1`] = `
<styled.ul
  options={
    Array [
      Object {
        "label": "Option One",
        "onClick": [Function],
      },
      Object {
        "disabled": true,
        "label": "Option Two",
        "onClick": [Function],
      },
      Object {
        "label": "Option Three",
        "onClick": [Function],
      },
    ]
  }
  theme={Object {}}
>
  <MenuOption
    key="0"
    label="Option One"
    onClick={[Function]}
  />
  <MenuOption
    disabled={true}
    key="1"
    label="Option Two"
    onClick={[Function]}
  />
  <MenuOption
    key="2"
    label="Option Three"
    onClick={[Function]}
  />
</styled.ul>
`;
